modules/ossm-cr-example.adoc

```diff
@@ -18,63 +18,257 @@ This example `ServiceMeshControlPlane` definition contains all of the supported
+|`3scale`, `grafana`, `jaeger`, `kiali`, `prometheus`, and `stackdriver`
+
+|cluster
+|Cluster is the general configuration of the cluster (cluster name, network name, multi-cluster, mesh expansion, etc.)
+|`meshExpansion`, `multiCluster`, `name`, and `network`
+
+|gateways
+|Gateways configures gateways for the mesh.
+|`enabled`, `additionalEgress`, `additionalIngress`, `egress`, `ingress`, and  `openshiftRoute`
+
+|general
+|General represents general control plane configuration that does not fit anywhere else.
+|`logging` and `validationMessages`
+
+|policy
+|Policy configures policy checking for the control plane. If `Values.policy.enabled` is set to true, policy checking is enabled.
+|`mixer`, `remote`, and `type`
+
+|profiles
+|Profiles selects the profile to use for default values.
+|`default`
+
+|proxy
+|Proxy configures the default behavior for sidecars. Many values were previously exposed through `Values.global.proxy`.
+|`accessLogging`, `adminPort`, `concurrency`, and `envoyMetricsService`
+
+|runtime
+|Runtime configuration for the control plane components.
+|`components`, and `defaults`
+
+|security
+|Security configures aspects of security for the control plane.
+|`certificateAuthority`, `controlPlane`, `identity`, `mtls`, and `trust`
+
+|techPreview
+|TechPreview contains switches for features that are not GA yet.
+|N/A
+
+|telemetry
+|Telemetry configures telemetry for the mesh. `Values.mixer.telemetry.enabled`, set to true true.
+|`mixer`, `remote`, and `type`
+
+|tracing
+|Tracing configures tracing for the mesh.
+|`sampling`, `type`
+
+|version
+|Version specifies what Maistra version of the control plane to install. When creating a `ServiceMeshControlPlane` with an empty version, the admission webhook sets the version to the current version. Existing `ServiceMeshControlPlanes` with an empty version are treated as having the version set to "v1.0"
+|string
+|===
+
+ControlPlaneStatus represents the current state of a `ServiceMeshControlPlane`.
+
+.`ServiceMeshControlPlane` resource `ControlPlaneStatus`
+|===
+|Name |Description |Type
+
+|annotations
+|Annotations is an unstructured key value map used to store additional, usually redundant status information, such as the number of components deployed by the `ServiceMeshControlPlane`. The reason to add this redundant information is to make it available to kubectl, which does not yet allow counting objects in JSONPath expressions.
+|map[string]string
+
+|conditions
+Represents the latest available observations of the object’s current state.
+|string
+
+|components
+|N/A
+|string
+
+|appliedSpec
+|The resulting specification of the configuration options after all profiles have been applied.
+|`ControlPlaneSpec`
+
+|appliedValues
+|The resulting values.yaml used to generate the charts.
+|`ControlPlaneSpec`
+
+|chartVersion
+|The version of the charts that were last processed for this resource.
+|string
+
+|observedGeneration
```

Comment:
@rcernich this isn't configurable and shouldn't matter to the user right?

Reply:
Same with chartVersion, operatorVersion, and appliedValues?

Reply:
This is important as it lets the user figure out if the SMCP has been reconciled and the status pertains to the correct spec version. For example, when users update the SMCP.spec, the SMCP.status will show condition `Reconciled` as `true` for a while - before the operator starts to reconcile the SMCP. This may lead the user to think that the operator has already reconciled the user's changes, which is not true. This is only true when the `observedGeneration` matches `metadata.generation`.

This is one of the biggest caveats when inspecting the status of a resource. Ideally, the Reconciled condition would change to false the exact moment when you update the spec, but that's not really possible.  You might think that an admission webhook should do this, but it can't, since the UPDATE request is modifying the metadata/spec part of the resource and you can't update the status in the same operation (the webhook would have to post the new status to the status subresource).